integration_tests/tests/self_test.rs

```diff
@@ -1,10 +1,80 @@
 use std::net::{TcpListener, TcpStream};
 
-use simple_https_client::tls13client;
+use simple_https_client::{
+    tls13client, SHA256_Aes128Gcm_EcdsaSecp256r1Sha256_P256,
+    SHA256_Aes128Gcm_EcdsaSecp256r1Sha256_X25519, SHA256_Aes128Gcm_RsaPssRsaSha256_P256,
+    SHA256_Aes128Gcm_RsaPssRsaSha256_X25519, SHA256_Chacha20Poly1305_EcdsaSecp256r1Sha256_P256,
+    SHA256_Chacha20Poly1305_EcdsaSecp256r1Sha256_X25519,
+    SHA256_Chacha20Poly1305_RsaPssRsaSha256_P256, SHA256_Chacha20Poly1305_RsaPssRsaSha256_X25519,
+    SHA384_Aes256Gcm_EcdsaSecp256r1Sha256_P256, SHA384_Aes256Gcm_EcdsaSecp256r1Sha256_X25519,
+    SHA384_Aes256Gcm_RsaPssRsaSha256_P256, SHA384_Aes256Gcm_RsaPssRsaSha256_X25519,
+};
 use simple_https_server::tls13server;
 
 #[test]
-fn self_test() {
+#[should_panic]
+fn test_sha256_chacha20_poly1305_rsa_pss_rsa_sha256_x25519() {
+    self_test_algorithm(SHA256_Chacha20Poly1305_RsaPssRsaSha256_X25519);
+}
+#[test]
+fn test_sha256_chacha20_poly1305_ecdsa_secp256r1_sha256_x25519() {
+    self_test_algorithm(SHA256_Chacha20Poly1305_EcdsaSecp256r1Sha256_X25519);
+}
+#[test]
+fn test_sha256_chacha20_poly1305_ecdsa_secp256r1_sha256_p256() {
+    self_test_algorithm(SHA256_Chacha20Poly1305_EcdsaSecp256r1Sha256_P256);
+}
+#[test]
+#[should_panic]
+fn test_sha256_chacha20_poly1305_rsa_pss_rsa_sha256_p256() {
+    self_test_algorithm(SHA256_Chacha20Poly1305_RsaPssRsaSha256_P256);
+}
+#[test]
+#[cfg(not(target_os = "macos"))]
```

Comment:
This should work on macos as well. It doesn't work on the macos GH runners and ARM-based macs. But it works on most other macs.
I don't think there's a nice way to disable this with a config, but you can only run the test when `libcrux_platform::aes_ni_support` is true (that crate is in the libcrux repository as well).